var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -1,103 +1,112 @@
-                open('.bindir', 'w').write(bindir)
-
-                if not os.path.isfile(special):
-                    raise Exception("Failed to build neurodamus core mods")
-
-                # Cleanup
-                for f in find(bindir, '*.lo'): os.remove(f)
-                for f in find(bindir, '*.mod'): os.remove(f)
-                for f in find(bindir + '/.libs', '*.o'): os.remove(f)
-                os.mkdir(bindir + "/modc")
-                for f in find(bindir, "*.c*"): shutil.move(f, bindir + "/modc/")
+        # '+common' will bring common mods.
+        # Otherwise build purely neurodamus helper mechs
+        if self.spec.satisfies('+common'):
+            copy_all('resources/common/hoc', "hoc")
+            copy_all('resources/common/mod', "mod")
+        else:
+            # These two mods are also part of common
+            os.remove("mod/VecStim.mod")
+            os.remove("mod/netstim_inhpoisson.mod")
+
+        # If we shall build mods for coreneuron, only bring from core those specified
+        if self.spec.satisfies("+coreneuron"):
+            mkdirp("mod_core")
+            with open(join_path("mod", _CORENRN_MODLIST_FNAME)) as core_mods:
+                for aux_mod in core_mods:
+                    modpath = join_path("mod", aux_mod.strip())
+                    if os.path.isfile(modpath):
+                        shutil.copy(modpath, 'mod_core')
+
+    def build(self, spec, prefix):
+        """ Build mod files from with nrnivmodl / nrnivmodl-core.
+            To support shared libs, nrnivmodl is also passed RPATH flags.
+        """
+        variant_to_compile_flag = {
+            "~mpi": "-DDISABLE_MPI",
+            "~hdf5": "-DDISABLE_HDF5",
+            "~reportinglib": "-DDISABLE_REPORTINGLIB",
+            "+synapsetool": "-DENABLE_SYNTOOL"
+        }
+
+        compile_flags = " ".join(flag for variant, flag in variant_to_compile_flag.items()
+                                 if spec.satisfies(variant))
+        self._build_mods('mod', '', compile_flags, 'mod_core')
+
+    # NOTE: install() is inherited
 
     def setup_environment(self, spack_env, run_env):
-        run_env.set('HOC_LIBRARY_PATH', self.prefix.hoc)
-        if self.spec.satisfies('+common'):
-            run_env.set('MOD_LIBRARY_PATH', self.prefix.mod)
-            bindir_info = self.prefix.join('.bindir')
-            if os.path.isfile(bindir_info):
-                bindir = open(bindir_info, 'r').readline()
-                mechlib = find_libraries('libnrnmech', join_path(self.prefix, bindir, '.libs'))[0]
-                run_env.set('NRNMECH_LIB_PATH', mechlib)
-                run_env.set('BGLIBPY_MOD_LIBRARY_PATH', mechlib)
-            else:
-                tty.warn("No .bindir info file found. NRNMECH_LIB_PATH env var wont be set")
+        self._setup_environment_common(spack_env, run_env)
+        for lib in find(self.prefix.lib, 'libnrnmech*'):
+            run_env.set('NRNMECH_LIB_PATH', lib)
```

Comment:
if we are `setting single` env var, how this works in loop?

Reply:
We will find 0 (build time) or 1 library (module creation time) so the for is a quick IF with var storage (before python 3.8 :)